ngraph/src/ngraph/op/prior_box.cpp

```diff
@@ -145,3 +146,54 @@ bool op::PriorBox::visit_attributes(AttributeVisitor& visitor)
     visitor.on_attribute("attrs.scale_all_sizes", m_attrs.scale_all_sizes);
     return true;
 }
+
+namespace
+{
+    template <element::Type_t ET>
+    bool evaluate(const HostTensorPtr& arg0,
+                  const HostTensorPtr& arg1,
+                  const HostTensorPtr& out,
+                  op::PriorBoxAttrs attrs)
+    {
+        runtime::reference::prior_box(arg0->get_data_ptr<ET>(),
+                                      arg1->get_data_ptr<ET>(),
+                                      out->get_data_ptr<float>(),
+                                      out->get_shape(),
+                                      attrs);
+        return true;
+    }
+
+    bool evaluate_prior_box(const HostTensorPtr& arg0,
+                            const HostTensorPtr& arg1,
+                            const HostTensorPtr& out,
+                            const op::PriorBoxAttrs& attrs)
+    {
+        bool rc = true;
+        switch (arg0->get_element_type())
+        {
+            TYPE_CASE(i8)(arg0, arg1, out, attrs);
+            break;
+            TYPE_CASE(i16)(arg0, arg1, out, attrs);
+            break;
+            TYPE_CASE(i32)(arg0, arg1, out, attrs);
+            break;
+            TYPE_CASE(i64)(arg0, arg1, out, attrs);
+            break;
+            TYPE_CASE(u8)(arg0, arg1, out, attrs);
+            break;
+            TYPE_CASE(u16)(arg0, arg1, out, attrs);
+            break;
+            TYPE_CASE(u32)(arg0, arg1, out, attrs);
+            break;
+            TYPE_CASE(u64)(arg0, arg1, out, attrs);
+            break;
```

Comment:
@itikhono, is it really required to instantiate so many versions depending on type? Have you measured the binary size increase?

Reply:
@slyalin Any of these types will be valid for this operation. Using the current approach, we must specify all of these types here. Not measured.

Reply:
As far as I know, the PriorBox input parameters are tensor shapes. Using `i8` or `u8` type for it looks strange. IMHO, the operation should support only single element type following from the type used to store tensor shape (`i64` I guess).